### PR TITLE
copy from prototype

### PR DESCRIPTION
**Pull request by AGeck113**

### CODE REVIEW

File: components/AddEventForm/index.js [hunk 1]
```diff
@@ -31,9 +89,29 @@ export default function AddEventForm({ onSubmit, appointment }) {
...
-        <select
+      <StyledLabel htmlFor="title"> Title:</StyledLabel>
+      <StyledInput
+        id="title"
+        required
+        maxLength={20}
+        type="text"
+        name="title"
+        defaultValue={appointment ? appointment.title : null}
+      ></StyledInput>
```
Comment (NikRoe) [inline components/AddEventForm/index.js:100]: inputs are self closing:

```suggestion
      <StyledInput
        id="title"
        required
        maxLength={20}
        type="text"
        name="title"
        defaultValue={appointment ? appointment.title : null}
      />
```

File: components/AddEventForm/index.js [hunk 2]
```diff
@@ -12,14 +13,71 @@ export default function AddEventForm({ onSubmit, appointment }) {
...
+    background-color: lightgray;
+  `;
+  const StyledDateInput = styled.input`
+    width: 95%;
+    margin: 0.5 auto;
+    border-radius: 999px;
+    height: 2rem;
+    text-align: center;
+    background-color: lightgray;
+  `;
```
Comment (NikRoe) [inline components/AddEventForm/index.js:73]: those inputs look almost identical. Is it possible to have only one styled component you use in both places?

File: components/AddEventForm/index.js [hunk 3]
```diff
@@ -44,68 +122,45 @@ export default function AddEventForm({ onSubmit, appointment }) {
...
           max={999999}
           name="cost"
           defaultValue={appointment ? appointment.cost : null}
-        ></input>
-        €
-      </label>
-      <label>
-        Priority:
-        <select
+        ></StyledInputCost>
```
Comment (NikRoe) [inline components/AddEventForm/index.js:135]: inputs are self-closing:

```suggestion
        />
```

File: components/Documents/index.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  position: absolute;
+  z-index: 10;
+  right: -1.2rem;
+  background-color: hsla(0, 93%, 40%, 0.89);
+  border-radius: 999px;
+`;
+export default function Documents({ documents, onDelete }) {
+  return documents.map((document) => {
+    return (
+      <StyledDocument key={nanoid()}>
```
Comment (NikRoe) [inline components/Documents/index.js:35]: You create a new key with every new render of this component and this has negative effects on the performance. Keys should be stable. For more information, you can see [this link](https://reactjs.org/docs/lists-and-keys.html#keys).

Please work with `document._id` instead.

File: components/EditCarForm/index.js [hunk 5]
```diff
@@ -95,18 +171,26 @@ export default function EditCarForm({ onSubmit, initialValues }) {
...
+              })}
+            </StyledFieldset>
+          );
+        })}
+        <StyledButtonContainer>
+          {initialValues ? (
+            <StyledCancelButton type="button" onClick={onCancel}>
+              <SVGIcon variant="cancel" width="50px" />
+            </StyledCancelButton>
+          ) : null}
```
Comment (NikRoe) [inline components/EditCarForm/index.js:188]:
```suggestion
          {initialValues && (
            <StyledCancelButton type="button" onClick={onCancel}>
              <SVGIcon variant="cancel" width="50px" />
            </StyledCancelButton>
          )}
```

File: components/LinkContainer/Index.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+const StyledParagraph = styled.p`
+  color: lightgray;
+  font-size: 1.1rem;
+  margin-top: 0.4rem;
+  font-weight: bold;
+`;
+export default function LinkSection() {
+  return (
+    <LinkContainer>
+      <StyledLink href={`/events/latest`}>
```
Comment (NikRoe) [inline components/LinkContainer/Index.js:18]: you don't need brackets and template literals here

```suggestion
      <StyledLink href="/events/latest">
```

File: components/LinkContainer/Index.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        <SVGIcon variant="last" width="3.5rem" />
+        <StyledParagraph>Last</StyledParagraph>
+      </StyledLink>
+      <StyledLink href={`/events/upcoming`}>
```
Comment (NikRoe) [inline components/LinkContainer/Index.js:22]: you don't need brackets and template literals here

```suggestion
      <StyledLink href="/events/upcoming">
```

File: components/LinkContainer/Index.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        <SVGIcon variant="next" width="3.5rem" />
+        <StyledParagraph>Next</StyledParagraph>
+      </StyledLink>
+      <StyledLink href={`/events/wishlist`}>
```
Comment (NikRoe) [inline components/LinkContainer/Index.js:26]: you don't need brackets and template literals here

```suggestion
      <StyledLink href="/events/wishlist">
```

File: components/LinkContainer/Index.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+        <SVGIcon variant="wish" width="3.5rem" />
+        <StyledParagraph>Wishes</StyledParagraph>
+      </StyledLink>
+      <StyledLink href={`/map`}>
```
Comment (NikRoe) [inline components/LinkContainer/Index.js:30]: you don't need brackets and template literals here

```suggestion
      <StyledLink href="/map">
```

File: pages/api/userCars/[id].js [hunk 10]
```diff
@@ -0,0 +1,27 @@
+//even needed now?
```
Comment (NikRoe) [inline pages/api/userCars/[id].js:1]: do you need this file?

File: pages/createCar/index.js [hunk 11]
```diff
@@ -90,25 +126,32 @@ export default function CreateCar() {
...
+        <StyledSubmitButton type="submit">
+          <SVGIcon variant="searchCar" width="30px" />
+        </StyledSubmitButton>
+      </StyledForm>
       {searchFailed ? (
         <p>
           Sorry, we can not find the Vin in our data. Please control your vin.
           If you want, you can also use the form to register your car manually.
         </p>
       ) : null}
```
Comment (NikRoe) [inline pages/createCar/index.js:154]:
```suggestion
      {searchFailed && (
        <p>
          Sorry, we can not find the Vin in our data. Please control your vin.
          If you want, you can also use the form to register your car manually.
        </p>
      )}
```

File: pages/events/[type].js [hunk 12]
```diff
@@ -31,6 +35,9 @@ export default function EventsPage() {
   if (!session) {
     return <Login />;
   }
+  if (type != "wishlist" && type != "upcoming" && type != "latest") {
```
Comment (NikRoe) [inline pages/events/[type].js:38]: is there a reason you don't work with strict inequality here? 🤔 

```suggestion
  if (type !== "wishlist" && type !== "upcoming" && type !== "latest") {
```

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hi Andreas,

looks really nice and I left only a few notes. Please take a look at the suggestions and questions I left you.

File: components/AddEventForm/index.js (hunk 1 above)
Comment (AGeck113) [reply components/AddEventForm/index.js:100]: Done

File: components/AddEventForm/index.js (hunk 2 above)
Comment (AGeck113) [reply components/AddEventForm/index.js:73]: Done

File: components/AddEventForm/index.js (hunk 3 above)
Comment (AGeck113) [reply components/AddEventForm/index.js:135]: Done

File: components/Documents/index.js (hunk 4 above)
Comment (AGeck113) [reply components/Documents/index.js:35]: Done

File: components/EditCarForm/index.js (hunk 5 above)
Comment (AGeck113) [reply components/EditCarForm/index.js:188]: Thank you, done

File: pages/api/userCars/[id].js (hunk 10 above)
Comment (AGeck113) [reply pages/api/userCars/[id].js:1]: Deleted it already last time, somehow it has found his way back 😅

File: pages/createCar/index.js (hunk 11 above)
Comment (AGeck113) [reply pages/createCar/index.js:154]: Thanks, done

File: pages/events/[type].js (hunk 12 above)
Comment (AGeck113) [reply pages/events/[type].js:38]: Yes, the reason is that I forgot to use it 😄 Done

File: pages/api/userCars/[id].js (hunk 10 above)
Comment (NikRoe) [reply pages/api/userCars/[id].js]: sneaky file, if it happens again, reach out to us :)

Comment (AGeck113) [reply pages/api/userCars/[id].js]: Thanks, i will if it keeps coming back

**Review by NikRoe — APPROVED** [review]
Very nice 👍

**Pull request merged**